CMakeLists.txt

```diff
@@ -436,7 +441,21 @@ add_subdirectory(coreneuron)
 if(CORENRN_ENABLE_GPU)
   get_target_property(CORENRN_LINK_LIBRARIES coreneuron INTERFACE_LINK_LIBRARIES)
   foreach(LIB ${CORENRN_LINK_LIBRARIES})
-    set_property(GLOBAL APPEND_STRING PROPERTY CORENEURON_LIB_LINK_FLAGS " ${LIB}")
+    get_filename_component(dir_path ${LIB} DIRECTORY)
+    if(TARGET ${LIB})
+      # See, for example, caliper where the coreneuron target depends on the caliper target (so we
+      # get LIB=caliper in this loop), but -l and -L are already added manually here:
+      # https://github.com/BlueBrain/CoreNeuron/blob/856cea4aa647c8f2b0d5bda6d0fc32144c5942e3/CMakeLists.txt#L411-L412
+      message(
+        NOTICE
+        "Ignoring dependency '${LIB}' of 'coreneuron' and assuming relevant flags have already been added to CORENEURON_LIB_LINK_FLAGS."
+      )
```

Comment:
It's true for calliper and I assume this works as before when there are other libraries like `reportinglib`, `sonata` are linked?

Reply:
I think `reportinglib` doesn't hit this branch because it isn't a CMake target, just the name of a library. `sonata` seems to be the same, I see:
```
-- Reporting           | ON
--   sonatareport_INC  | /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_intel-2021.4.0-skylake/libsonata-report-1.0.0.20220218-ldqdrn/include
--   sonatareport_LIB  | /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_intel-2021.4.0-skylake/libsonata-report-1.0.0.20220218-ldqdrn/lib/libsonatareport.so
--   reportinglib_INC  | /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_intel-2021.4.0-skylake/reportinglib-2.5.6-gdhqyp/include
--   reportinglib_LIB  | /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_applications/install_intel-2021.4.0-skylake/reportinglib-2.5.6-gdhqyp/lib64/libreportinglib.so
```
in the configure output from the CI on this PR.

This is still clearly fragile, but it's a bigger project to fix this logic robustly (and consistently across NEURON and CoreNEURON).

Reply:
ok, thanks for clarification. 👍